src/code_translation/ir/Optimizer.py

```diff
@@ -36,8 +37,11 @@ def map_old_to_new_nodes(self):
             if isinstance(ir_node, DataFrameNode):
                 for target, value in self.variables.items():
                     if ir_node == value:
-                        cst_translation = ir_node.to_cst_translation()
-                        # TODO: Needs awareness of assignment name
+                        if ir_node.library in self.sql_access_methods:
+                            sql_access_method = self.sql_access_methods[ir_node.library]
+                        else:
+                            raise Exception(f"No sql access method for {ir_node.library}")
+                        cst_translation = ir_node.to_cst_translation(sql_access_method)
```

Comment:
Yeah I don't really this. We
- define the sql_access_methods in the InputModules (although this is more of an output thing)
- Have a method inside the NodeSelector which gathers all the information from the different InputModules (A method that is never used inside the NodeSelector itself; again, because this is an output thing)
- We call this method from the Orchestrator and inject the result into the optimizer
- We find out the specific access method of the library of an IR Node and then inject this back into the IR Node's method for cst translation

I feel like this is not very maintainable, scalable or modular whatsoever